src/formats/chr.c

```diff
@@ -285,6 +286,15 @@ int sd_chr_save(sd_chr_file *chr, const char *filename) {
     return SD_SUCCESS;
 }
 
+void sd_chr_filename(const sd_chr_file *chr, char filename[static 9]) {
+    size_t len = min2(strlen(chr->pilot.name), 8);
+    for(size_t i = 0; i < len; ++i) {
+        unsigned char c = chr->pilot.name[i];
+        filename[i] = isalnum(c) ? toupper(c) : '_';
```

Comment:
same critique as before with strtoupper. cast the potentially-signed `char` to `unsigned char` before passing to isalnum and toupper to avoid UB.